src/mds/CDir.cc

```diff
@@ -1634,9 +1638,11 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
      * doesn't exist) but for which no explicit negative dentry is in
      * the cache.
      */
-    CDentry *dn = NULL;
-    if (!stale)
-      dn = lookup(dname, last);
+    CDentry *dn;
+    if (stale)
+	    dn = lookup_exact_snap(dname, last);
+    else
+	    dn = lookup(dname, last);
```

Comment:
Looks like you accidentally busted the indentation here (kernel standards instead of Ceph userspace standards?).